Files/Filesystem/StorageItems/StreamWithContentType.cs

```diff
@@ -10,74 +10,98 @@ namespace Files.Filesystem.StorageItems
 {
     public class InputStreamWithDisposeCallback : IInputStream
     {
-        private IInputStream stream;
+        private Stream stream;
+        private IInputStream iStream;
         public Action DisposeCallback { get; set; }
 
         public InputStreamWithDisposeCallback(Stream stream)
         {
-            this.stream = stream.AsInputStream();
+            this.stream = stream;
+            this.iStream = stream.AsInputStream();
         }
 
         public IAsyncOperationWithProgress<IBuffer, uint> ReadAsync(IBuffer buffer, uint count, InputStreamOptions options)
         {
-            return stream.ReadAsync(buffer, count, options);
+            return iStream.ReadAsync(buffer, count, options);
```

Comment:
`stream.AsInputStream().ReadAsync(...)`?

Reply:
In the constructor I dot this `this.iStream = stream.AsInputStream();` to avoid calling AsInputStream() at every read operation.